### PR TITLE
First attempt to finish js-dom task

### PR DESCRIPTION
**Pull request by vitaliykravchyk**
_Complete the form below and remove this line._

### DOM

[Demo](https://vitaliykravchyk.github.io/kottansDOM/) |
[Code base](https://github.com/vitaliykravchyk/kottans-frontend/tree/master/task_js_dom)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/vitaliykravchyk/js-dom/js/main.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+var Luntik = {
+    data: [
```
Comment (AMashoshyna) [inline submissions/vitaliykravchyk/js-dom/js/main.js]: Using `var` is technically not deprecated, but `let` and `const` are preferable

File: submissions/vitaliykravchyk/js-dom/js/main.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        {name: "Пчелёнок", image: "img/bee.jpg", description: "Пчелёнок — друг Лунтика, который ходит в школу пчёл и учится базовым навыкам пчелы. Пчелёнок трудолюбив и любознателен, но при этом, очень стеснителен. Он раньше считал, что ему незачем ходить в школу, ведь он и так всё знает, но после серии «Прогулял» он понял, что знания, полученные в школе, пригодятся. Самый умный среди своих детских персонажей. Иногда рассказывает своим друзьям о том, что он сегодня узнал. Мечта Пчелёнка — стать героем и спасти мир. Во время событий сериала смог перебороть страх темноты."}
+    ],
+    init: function(){
+        let createItemFunction = this.createItem.bind(this);
+        let removeContentFunction = this.removeContent.bind(this);
+        let removeStylesFunction = this.removeStyles.bind(this);
+        document.querySelector('#friend-list').addEventListener('click', function(){
+            removeContentFunction();
```
Comment (AMashoshyna) [inline submissions/vitaliykravchyk/js-dom/js/main.js]: You do not need to bind all of your methods. It its enough to bind only the function you pass to `addEventListener`. I suggest you put this anonymous function into another method called something like  `changeItem` and bind it.

File: submissions/vitaliykravchyk/js-dom/js/main.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+    },
+    removeStyles: function(){
+        for(let i = 0; i < document.querySelectorAll('#friend-list li').length;i++){
+            if(document.querySelectorAll('#friend-list li')[i].classList.contains('person')){
+                document.querySelectorAll('#friend-list li')[i].classList.remove('person');
+            }
+        }
+    }
+}
```
Comment (AMashoshyna) [inline submissions/vitaliykravchyk/js-dom/js/main.js]: Please add newline at the end of file

File: submissions/vitaliykravchyk/js-dom/js/main.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        if(event.target.hasAttribute('data-item')){            
+            if(document.querySelector('main .content').childNodes.length > 0){
+                if(document.querySelector('main .content').childNodes[0].classList.contains('infoBlock')){
+                    document.querySelector('main .content').removeChild(document.querySelector('.infoBlock'));
+                };
+            }
+        }
+    },
+    removeStyles: function(){
+        for(let i = 0; i < document.querySelectorAll('#friend-list li').length;i++){
```
Comment (AMashoshyna) [inline submissions/vitaliykravchyk/js-dom/js/main.js]: In this method  on every loop iteration you make same DOM query at least two times:
- line 50
- line 51
- line 52 (sometimes)
DOM manipulations are costly in terms of performance. You can put the result of the first query into a variable and use it.
Also, in this particular case you can simply query by `.person`.

**Comment by AMashoshyna** [comment]
Great job! Please see my comments to you code.

**Comment by vitaliykravchyk** [comment]
Hi check my fix pls, ty)

**Comment by AMashoshyna** [comment]
Looks much better. But there is one tricky thing.
Please check your code in Firefox and see what happens. Hint: problem is in `changeItem` method.
Let me know, if you need help.

**Comment by vitaliykravchyk** [comment]
Now it's works in Firefox

**Comment by AMashoshyna** [comment]
Great!
It will merge the PR, but as an enhancement consider the following.
```
  document.querySelector('#friend-list').addEventListener('click', function(event){
            changeItemFunction(event)
        }); 
```
Since `changeItemFunction` accepts one parameter, and second parameter of `addEventListener` will be provided with one argument, you can write it shorter:
```
  document.querySelector('#friend-list').addEventListener('click', changeItemFunction); 
```

**Pull request merged**